dist/centrifuge.d.ts

```diff
@@ -0,0 +1,115 @@
+
+import EventEmitter from 'events';
+
+export = Centrifuge;
+
+declare class Centrifuge extends EventEmitter {
+    constructor(url: string, options?: Centrifuge.Options);
+    setToken(token: string): void;
+    setConnectData(data: any): void;
+    rpc(data: any): Promise<any>;
+    send(data: any): Promise<{}>;
+    publish(channel: string, data: any): Promise<{}>;
+    getSub(channel: string): Centrifuge.Subscription;
+    isConnected(): boolean;
+    connect(): void;
+    disconnect(): void;
+    ping(): void;
+    startBatching(): void;
+    stopBatching(): void;
+    startSubscribeBatching(): void;
+    stopSubscribeBatching(): void;
+    subscribe(channel: string, events?: (...args: any[]) => void): Centrifuge.Subscription;
+    subscribe(channel: string, events?: Centrifuge.SubscriptionEvents): Centrifuge.Subscription;
+}
+
+declare namespace Centrifuge {
+    export interface Options {
+        debug?: boolean;
+        sockjs?: any;
+        promise?: any;
+        minRetry?: number;
+        maxRetry?: number;
+        timeout?: number;
+        ping?: boolean;
+        pingInterval?: number;
+        pongWaitTimeout?: number;
+        privateChannelPrefix?: string;
+        onTransportClose?: boolean;
+        sockjsServer?: string | null;
+        sockjsTransports?: string[];
+        refreshEndpoint?: string;
+        refreshHeaders?: object;
+        refreshParams?: object;
+        refreshData?: object;
+        refreshAttempts?: number | null;
+        refreshInterval?: number;
+        onRefreshFailed?: () => void;
+        onRefresh?: (context: object, cb: (resp: any) => void) => void;
+        subscribeEndpoint?: string;
+        subscribeHeaders?: object;
+        subscribeParams?: object;
+        subRefreshInterval?: number;
+        onPrivateSubscribe?: (message: {data: SubscribePrivateContext}, cb: (resp: any) => void) => void;
+    }
+    export class Subscription extends EventEmitter {
+        channel: string;
+        ready(callback: (context: SubscribeSuccessContext) => void, errback: (context: SubscribeErrorContext) => void): void;
+        subscribe(): void;
+        unsubscribe(): void;
+        publish(data: any): Promise<any>;
+        presence(): Promise<any>;
+        presenceStats(): Promise<any>;
+        history(): Promise<any>;
+    }
+    export interface SubscriptionEvents {
+        publish?: (message: PublishMessage) => void;
+        join?: (message: JoinLeaveMessage) => void;
+        leave?: (message: JoinLeaveMessage) => void;
+        subscribe?: (context: SubscribeSuccessContext) => void;
+        error?: (errContext: SubscribeErrorContext) => void;
+        unsubscribe?: (context: UnsubscribeContext) => void;
+    }
+    export interface PublishMessage {
+        uid: string;
+        data: any;
+        client?: string;
+        info?: MessageInfo;
+    }
+    export interface MessageInfo {
+        user? : string;
+        client? : string;
+        default_info?: object;
```

Comment:
better to do `any` here at in similar places - as in Protobuf case this fields will contain `uint8array` instead of object.

Reply:
```ts
    export interface MessageInfo {
        user? : string;
        client? : string;
        default_info?: object | Uint8Array;
        channel_info?: object | Uint8Array;
    }
```
It would be nice instead `object` use interface for `default_info` and `channel_info` data format.

Reply:
It does not have any interface as those fields are fully controlled by application developer and can contain any JSON (not only `object` btw but `any`).